taf/api/roles.py

```diff
@@ -438,9 +438,10 @@ def _enter_roles_infos(keystore: Optional[str], roles_key_infos: Optional[str])
             "Enter keystore location if keys should be loaded from or generated to keystore files (leave empty otherwise): "
         )
         if len(keystore):
-            if not Path(keystore).is_dir():
-                taf_logger.info(f"{keystore} does not exist")
-                keystore = None
+            keystore_path = Path(keystore)
```

Comment:
I had to think about this and I'd say that the new requirements were in conflict with some of our old requirements, so this might require a more work. One old requirement was to have the option to generate new keys and print them instead of saving them to files. This was supposed to enhance security, since you'd copy that value to a secure password manager. At the moment, we store root and target keys on Yubikeys, and the remaining ones (snapshot and timestamp) everyone stores inside keystore files. From the security standpoint, it is better to generate new keys every time, but at the moment, that is not the case. We want TAF to be able to be flexible enough so that it's up to the user what they want to do with their keys.
So let's do the following:
- If keys-description, or `roles_key_infos` how we call it here is not provided, ask the user to enter it. Do not ask the user about the keystore path at that time and do not print it at the end of that `_enter_roles_infos` function
- Now that we have `roles_key_infos`, check if all keys should be loaded from/stored to yubikeys
  - If that is not the case, check if keystore path is available (either read from keys-description.json or the keystore parameter)
  -  if keystore path is not specified, ask the user if they want to save/load keys from keystore files (make sure that the users enter y/n here, repeat until they provide an answer, instead of defaulting to one of the two answers)
  - If they their answer if negative,  do not raise an error, assume that the user wants the keys to entered using and printed to the command line. This use case should be tested in more detail, but is not a priority
 - if the keystore path is provided (regardless of how it was provided), check if the directory exists. 
   - Ideally, we could also iterate through all roles whose keys are not stored on yubikeys and check if the corresponding keystore files exist. Names of keystore files are based on names of the roles, so this can be automated. If some files are missing, print a message saying that keys of that and that role will be generated and ask them to confirm that that is what they want. This can be a future enhancement, so we can just create an issue now
   - if the directory does not exist, print that as a warning (use taf_logger) and ask the user if they want to enter a different path or generate new keys
   - Finally, once we have a keystore path that will be used going forward and if the user entered keys description through the command line, append that keystore path and then print the whole json

Reply:
I made the changes, but I had some questions:
1. If the user chooses NO to save/load keys from keystore files or choose a directory for the keystore files, can I use the current directory (Set keystore to ".")? Without a specified directory a user currently can’t generate keystore files.
2. If a keystore path is declared in keys-description and as a command argument and both are different, which should be taken?

Reply:
if the user does not want to save/load keys from the keystore files, we should just print the keys or ask the user to copy/paste the key into their terminal. I don't think that we've ever used that feature (or at least not recently), but it should all still be supported. Since this might require further changes, let's save the keystore files to `./keystore` for now. We should comment out that question though, it would be confusing
If keystore path is specified in both places, I think that the command argument should be used.

Reply:
Now that I'm testing this. Let's leave the question, I'll take this over and look into what's happening with printing/loading from the command line.